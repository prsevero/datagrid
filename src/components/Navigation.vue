<template>
    <nav>
        <router-link to="/"><img alt="Company Name Logo" src="@/assets/logo.svg"></router-link>
        <router-link to="/logout">Logout</router-link>
    </nav>
</template>


<script>
export default {
    name: 'Navigation',
};
</script>


<style scoped lang="sass">
nav
    align-items: center
    background: #fff
    bottom: 0
    box-shadow: 0 0 5px rgba(0, 0, 0, .3)
    display: flex
    height: 50px
    justify-content: space-between
    left: 0
    padding: 3px 0 3px 5px
    position: fixed
    right: 0
    top: 0
    z-index: 1
    a
        color: #333
        font-size: 1.3em
        font-weight: 600
        outline: none
        text-decoration: none
        &:focus, &:hover
            color: #555
        &:first-child
            height: 100%
        &:last-child
            padding: 5px
    img
        height: 100%
</style>
