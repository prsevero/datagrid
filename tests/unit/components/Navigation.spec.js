import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Navigation from '@/components/Navigation.vue';


const localVue = createLocalVue();
localVue.use(VueRouter);


describe('Navigation.vue', () => {

    let mountedNavigation;
    const navigation = () => {
        if (!mountedNavigation) {
            mountedNavigation = shallowMount(Navigation, {localVue});
        }
        return mountedNavigation;
    }

    beforeEach(() => {
        mountedNavigation = undefined;
    });


    // All tests will go here
    it('should match snapshot', () => {
        const wrapper = navigation();
        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it('always render a nav', () => {
        const wrapper = navigation();
        expect(wrapper.is('nav')).toBe(true);
    });

    it('the rendered navigation contains a home link', () => {
        const wrapper = navigation();
        expect(wrapper.findAll('router-link-stub[to="/"]').length).toBeGreaterThan(0);
    });

    it('the rendered navigation contains a logout link', () => {
        const wrapper = navigation();
        expect(wrapper.findAll('router-link-stub[to="/logout"]').length).toBeGreaterThan(0);
    });

});
