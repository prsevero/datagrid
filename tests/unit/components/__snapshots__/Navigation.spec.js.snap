// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation.vue should match snapshot 1`] = `
<nav>
  <router-link-stub
    event="click"
    tag="a"
    to="/"
  >
    <img
      alt="Company Name Logo"
      src="@/assets/logo.svg"
    />
  </router-link-stub>
   
  <router-link-stub
    event="click"
    tag="a"
    to="/logout"
  >
    Logout
  </router-link-stub>
</nav>
`;
